docs/VnVPlan/VnVPlan.tex

```diff
@@ -310,6 +310,83 @@ \subsubsection{Area of Testing2}
+
+\item{NFRT-SR2\\}
+
+Type: Functional, Dynamic
+
+Initial State: Data is stored in the application database.
+                
+Input: User will attempt to modify the data directly from the application database.
+                
+Output: Stored data are encrypted. Therefore, modifications to the data are denied.
+                
+How test will be performed: The team will attempt to use database tools to try to modify data within the application database. Then, the team will confirm that the application backend database is secure and has not been modified.
+
+\item{NFRT-SR3\\}
+
+Type: Dynamic, Penetration
+
+Initial State: Application is in operation.
+                
+Input: User will attempt simulated attacks like SQL injection, cross-site scripting, etc.
+                
+Output: Application will remain secure without any data breaches.
+                
+How test will be performed: The team will conduct a series of different types of simulated cyber attacks. The team will then observe application response and monitor application database.
+
+\item{NFRT-SR4\\}
+
+Type: Functional, Dynamic, Manual
+
+Initial State: Raw user data is stored in the application database.
+                
+Input: The raw user data is processed and then stored in the application database.
+                
+Output: The data is anonymized by data perturbation (inducing small amount of noise) and encryption. The data is also sanitized by data deletion or masking.
+                
+How test will be performed: The team will inspect the stored user data for any personal identifiers. The team will search for any presence of accessible sensitive user data.
+
+\item{NFRT-SR5\\}
+
+Type: Dynamic, Manual
+
+Initial State: Application has yet to collect user data.
+                
+Input: User will start using the application.
+                
+Output: A consent form will pop up before any user data is collected. No user data will be collected without explicit consent given by user.
+                
+How test will be performed: The team will check for prompts that ask users to consent to data collection. The team will then test the application with and without user consent to ensure compliance. The team will confirm no user data was collected and stored in the database without user consent.
+
+\item{NFRT-SR6\\}
+
+Type: 
+
+Initial State: Application is in operation.
+                
+Input: User will attempt various actions to induce system events.
+                
+Output: Activities and events will be accurately logged. The log can be reviewed by authorized administrators.
+                
+How test will be performed: The team will perform a variety of user actions and try to trigger system events. The team will verify if these actions and events are accurately logged to system log. The team will confirm that only authorized administrators can access these logs.
```

Comment:
Similar feedback with vague/ambiguous. Maybe target one aspect like adding a receipt or something and ensure logging happens there?

Reply:
resolved